### PR TITLE
feat: add home page

### PR DESCRIPTION
**Pull request by anik31**
## Home Page
I have added the home page. Ignore the index files. Focus only on the components, pages & context + reducer files.
 
## Files to check
- [ ]  src/App.js
- [ ]  src/components/Navbar/Navbar.jsx
- [ ]  src/components/VideoCard/VideoCard.jsx
- [ ]  src/components/VideoCard/video-card.css
- [ ]  src/context/video-context.js
- [ ]  src/hooks/useAsyncFetch.js
- [ ]  src/pages/home/CategoryCard.jsx
- [ ]  src/pages/home/Home.jsx
- [ ]  src/pages/home/home.css
- [ ]  src/reducer/video-reducer.js
 
 _PS - Ignore other files._
 
image
image
image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,41 +1,30 @@
...
-import logo from "./logo.png";
+import "./styles.css";
+import { Routes, Route } from "react-router-dom";
+import Mockman from "mockman-js";
+import {Home} from "./pages";
+import { useAsyncFetch } from "./hooks";
+import { Navbar } from "./components";
 
 function App() {
+  useAsyncFetch({
```
Comment (sainath7878) [inline src/App.js:9]: Appreciation: use of custom hook

**Review by sainath7878 — APPROVED** [review]
Everything looks good. Things I liked are good folder structure, use of custom hook for fetching data. UI looks awesome. Good work!

File: src/App.js (hunk 1 above)
Comment (anik31) [reply src/App.js:9]: thanks

File: src/pages/home/CategoryCard.jsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export function CategoryCard({value}){
+    const {categoryName,image} = value;
+    const navigate = useNavigate();
+
+    return (
+        <div 
+        className="category-card" 
+        onClick={()=>navigate("/explore")}>
+            <img src={image.src} alt={image.alt} loading="lazy" />
```
Comment (samyukthasudhakar) [inline src/pages/home/CategoryCard.jsx:11]: Appreciation : Like how you have added loading='lazy' for the image.

File: src/hooks/useAsyncFetch.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useVideos();
+
+    useEffect(()=>{
+        (async function asyncFetch(){
+            try{
+                const {data} = await axios.get(url);
+                dispatch({type:dispatchType, payload: data[dispatchPayload]});
+            }catch(err){
+                console.log(err);
```
Comment (samyukthasudhakar) [inline src/hooks/useAsyncFetch.js:14]: console.log statements can be removed.

File: src/hooks/useAsyncFetch.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { useVideos } from "../context/video-context";
+import {useEffect} from "react";
+import axios from "axios";
+
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useVideos();
```
Comment (samyukthasudhakar) [inline src/hooks/useAsyncFetch.js:6]: Like how you've made use of custom hooks to make things easier !

File: src/components/Navbar/Navbar.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { Link } from "react-router-dom";
+// import { useProducts } from "../../context/product-context";
+import "./navbar.css";
+
+export function Navbar(){
+    // const {state} = useProducts();
```
Comment (samyukthasudhakar) [inline src/components/Navbar/Navbar.jsx]: Can remove un-used code if not required.

**Review by samyukthasudhakar — APPROVED** [review]
Great work with the code. Love that it's neatly organized and is highly modular.

Also highly appreciate the thought process behind providing single commit with all the files to be reviewed. Made reviewing also easier. Will try to implement the same in my PR. Thanks !

File: src/pages/home/CategoryCard.jsx (hunk 2 above)
Comment (anik31) [reply src/pages/home/CategoryCard.jsx:11]: Thank you

File: src/hooks/useAsyncFetch.js (hunk 3 above)
Comment (anik31) [reply src/hooks/useAsyncFetch.js:14]: Currently I'm not using toast or showing error on UI, therefore I have used console.log. Will remove it soon.

File: src/hooks/useAsyncFetch.js (hunk 4 above)
Comment (anik31) [reply src/hooks/useAsyncFetch.js:6]: Thanks

File: src/components/Navbar/Navbar.jsx (hunk 5 above)
Comment (anik31) [reply src/components/Navbar/Navbar.jsx]: Thank you for pointing it out

**Pull request merged**